lib/TopStoriesFeed.jsm

```diff
@@ -401,29 +410,43 @@ this.TopStoriesFeed = class TopStoriesFeed {
     this.store.dispatch(ac.OnlyToOneContent(action, target));
   }
 
-  maybeAddSpoc(target) {
-    const updateContent = () => {
-      if (!this.shouldShowSpocs()) {
-        this.dispatchSpocDone(target);
-        return false;
-      }
-      if (Math.random() > this.spocsPerNewTabs) {
-        this.dispatchSpocDone(target);
-        return false;
-      }
-      if (!this.spocs || !this.spocs.length) {
-        // We have stories but no spocs so there's nothing to do and this update can be
-        // removed from the queue.
-        this.dispatchSpocDone(target);
+  filterSpocs() {
```

Comment:
I broke this out of `maybeAddSpoc` to a separate method to make it single purpose and therefore easier to isolate for unit testing. Now return values can be directly analyzed instead of having to dig deeper into other parts of the app's data flow. Also `maybeAddSpoc` was just feeling overloaded...